src/ProjectFileIO.cpp

```diff
@@ -211,6 +213,80 @@ class SQLiteIniter
    int mRc;
 };
 
+class SQLiteBlobStream final
+{
+public:
+   static std::optional<SQLiteBlobStream> Open(
+      sqlite3* db, const char* schema, const char* table, const char* column,
+      int64_t rowID, bool readOnly) noexcept
+   {
+      if (db == nullptr)
+         return {};
+
+      sqlite3_blob* blob;
+
+      const int rc = sqlite3_blob_open(
+         db, schema, table, column, rowID, readOnly ? 0 : 1, &blob);
```

Comment:
... just in case SQLITE_MISUSE is the return value, see the docs https://www.sqlite.org/c3ref/blob_open.html

Reply:
It's really confusing to be fair. Is it safe to close a misused blob?

>This means that, provided the API is not misused, it is always safe to call sqlite3_blob_close() on *ppBlob after this function it returns.

Looking into the source code - there is a single case where ppBlob will not be initialized and MISUSE is returned.